src/arrayMethodJoin.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    if (separator === '') {
+      return this.toString().replace(/,/gi, '');
+    }
+
+    if (separator === null) {
+      return this.toString().replace(/,/gi, null);
+    }
+
+    if (!separator) {
+      return this.toString();
+    }
+
+    return this.toString().replace(/,/gi, separator);
```

Comment:
```suggestion
    return this.toString().replace(/,/gi, separator);
```
В цьому завданні не можна користуватись іншими методами JS. Лише базовими циклами, іфами і конкатинацією/інтерполяцією.
Давай переробимо